### PR TITLE
metadata topic added

### PR DESCRIPTION
**Pull request by MonikaMrozek**
Added topic describing extending Saleor using metadata.

### CODE REVIEW

File: docs/api/metadata.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+title: Extending Models Using Metadata
+---
+
+Saleor gives you the possibility to customize your shop without the need of any changes in the code. Once you get your own Saleor instance, you can extend most of Saleor's database models using metadata. 
+
+For example, you can add specific fields to metadata of objects to store extra information on that object. Using metadata you can also customize the way your storefront appears to your customers. 
+
+The following models contain metafields which data can be modified:
+
+* `collection`
```
Comment (maarcingebala) [inline docs/api/metadata.md]: I would use `PascalCase` for model names, that is the convention in Django and in GraphQL when referring to objects.

File: docs/api/metadata.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
+---
+id: metadata
+title: Extending Models Using Metadata
+---
+
+Saleor gives you the possibility to customize your shop without the need of any changes in the code. Once you get your own Saleor instance, you can extend most of Saleor's database models using metadata. 
+
+For example, you can add specific fields to metadata of objects to store extra information on that object. Using metadata you can also customize the way your storefront appears to your customers.
```
Comment (maarcingebala) [inline docs/api/metadata.md]: Shouldn't it be:
```suggestion
For example, you can add specific fields to metadata of objects to store extra information on that object. By using metadata you can also customize the way your storefront appears to your customers. 
```

File: docs/api/metadata.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+                  {
+                    "name": "vatlayer",
+                    "metadata": [
+                      {
+                        "key": "code",
+                        "value": "standard"
+                      },
+                      {
+                        "key": "description",
+                        "value": ""
```
Comment (maarcingebala) [inline docs/api/metadata.md]: @korycins Can we add some example description here?

File: docs/api/metadata.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+```
+mutation {
+  productUpdateMetadata(id:"UHJvZHVjdDo3Mg==", input:{namespace:"my_namespace1", clientName: "custom_name", key:"special_product", value:"true"}){
+    errors{
+      field
+      message
+    }
+    product{
+		meta{
```
Comment (maarcingebala) [inline docs/api/metadata.md]: @korycins Please format the example queries properly. You can use "Prettify" option in the Playground. Not only this one but all of them, some of them have pretty long attributes lists.

File: docs/api/metadata.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+}
+```
+
+## Private and public metafields
+
+Models can have two meta fields:
+
+* `meta` - This field is public and visible for the non-authenticated user
+
+* `privateMeta` This  is protected and visible only for staff user with expected permissions
```
Comment (maarcingebala) [inline docs/api/metadata.md]:
```suggestion
* `privateMeta` This is protected and visible only for staff user with expected permissions
```

File: docs/api/metadata.md (hunk 4 above)
Comment (MonikaMrozek) [reply docs/api/metadata.md]: My bad! I've asked Maciek to paste the code snippets into the task in Jira, so I can place them in the docs. @korycins looks like you will have to redo them anyway :( sorry for the doubkle work!

**Pull request merged**